Source/Initialization/WarpXInitData.cpp

```diff
@@ -803,28 +803,35 @@ WarpX::InitLevelData (int lev, Real /*time*/)
     // provided in the input file.
     if (B_ext_grid_s == "parse_b_ext_grid_function") {
 
-#ifdef WARPX_DIM_RZ
-       WARPX_ABORT_WITH_MESSAGE(
-           "E and B parser for external fields does not work with RZ -- TO DO");
-#endif
-
-       //! Strings storing parser function to initialize the components of the magnetic field on the grid
-       std::string str_Bx_ext_grid_function;
-       std::string str_By_ext_grid_function;
-       std::string str_Bz_ext_grid_function;
+        //! Strings storing parser function to initialize the components of the magnetic field on the grid
+        std::string str_Bx_ext_grid_function;
+        std::string str_By_ext_grid_function;
+        std::string str_Bz_ext_grid_function;
 
+#ifdef WARPX_DIM_RZ
+        std::stringstream warnMsg;
+        warnMsg << "Parser for external B (r and theta) fields does not work with RZ\n"
+            << "The initial Br and Bt fields are currently hardcoded to 0.\n"
+            << "The initial Bz field should only be a function of z.\n";
+        ablastr::warn_manager::WMRecordWarning(
+          "Inputs", warnMsg.str(), ablastr::warn_manager::WarnPriority::high);
+        str_Bx_ext_grid_function = "0";
+        str_By_ext_grid_function = "0";
```

Comment:
As a side-note: if this is needed, I could try to finalize this PR:
https://github.com/ECP-WarpX/WarpX/pull/3913

Reply:
It would be helpful but not urgently. Specifically, I could see a situation in which we would like to apply an analytic perturbation to the axial field.